src/test/scala/nl.knaw.dans.easy.bagstore/component/BagStoreSpec.scala

```diff
@@ -167,4 +169,15 @@ class BagStoreSpec extends TestSupportFixture
     Files.getPosixFilePermissions(bagInStore.resolve("data")) shouldBe dirPermissions
     Files.getPosixFilePermissions(bagInStore.resolve("data/sub")) shouldBe dirPermissions
   }
+
+  it should "result in a failure when the fetch.txt is invalid" in {
+    val uuid1 = UUID.fromString("00000000-0000-0000-0000-000000000001")
+    val fetchFile = File(testBagPrunedB.resolve("fetch.txt"))
+
+    Files.write(fetchFile.path, fetchFile.contentAsString.replaceAll("01", "10").getBytes())
+
+    bagStore.add(testBagPrunedB, Some(uuid1)) should matchPattern {
+       case Failure(CompositeException(errors)) if errors.exists(_.isInstanceOf[IllegalArgumentException]) =>
```

Comment:
Could you make this test somewhat more precise, to see that it actually contains the excepted exception? Like, testing on the expected text.